zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -3070,3 +3095,116 @@ pub fn metadata_queries_exclude_unwanted_notes<T: ShieldedPoolTester, DSF, TC>(
         Some(5),
     );
 }
+
+#[cfg(feature = "pczt")]
+pub fn pczt_single_step<P0: ShieldedPoolTester, P1: ShieldedPoolTester, DSF>(
+    ds_factory: DSF,
+    cache: impl TestCache,
+) where
+    DSF: DataStoreFactory,
+    <DSF as DataStoreFactory>::AccountId: serde::Serialize + serde::de::DeserializeOwned,
+{
+    use zcash_protocol::consensus::ZIP212_GRACE_PERIOD;
+
+    let mut st = TestBuilder::new()
+        .with_data_store_factory(ds_factory)
+        .with_block_cache(cache)
+        .with_initial_chain_state(|_, network| {
+            // Initialize the chain state to after ZIP 212 became enforced.
+            let birthday_height =
+                network.activation_height(NetworkUpgrade::Nu5).unwrap() + ZIP212_GRACE_PERIOD;
```

Comment:
The grace period was after Canopy, not NU5. Do you mean this?
```suggestion
            let birthday_height = std::cmp::max(
                network.activation_height(NetworkUpgrade::Nu5).unwrap(),
                network.activation_height(NetworkUpgrade::Canopy).unwrap() + ZIP212_GRACE_PERIOD,
            );
```

Reply:
Initially I just activated NU5, but that didn't work because the NUs are too close together in regtest mode (they are actually activated at the same height). What matters here is that a) Orchard is available (so the test is general) which requires NU5, and b) ZIP 212 is enforced, which requires Canopy + `ZIP212_GRACE_PERIOD`. The current code ensures both hold at once.